goldigen/input_sanitizer.go

```diff
@@ -3,12 +3,18 @@ package main
 import "bytes"
 
 type sanitizer struct {
-	buf *bytes.Buffer
+	buf writer
 
 	inQuotes  bool
 	quoteChar byte
 }
 
+type writer interface {
+	WriteString(string) (int, error)
+	WriteByte(c byte) error
+	Bytes() []byte
+}
+
```

Comment:
I looked at these functions in the [`bytes.Buffer`][1] implementation. None of them actually have a code branch that can actually return an error. The only reason its defined as return argument is to implement the common interfaces of the standard library.

Therefore I think we should leave the unexported `sanitizer.buf` as it is instead of using an interface only to test error conditions that cannot actually occur. 

[1]: https://godoc.org/bytes#Buffer

Reply:
This was just to increase the coverage. There were cases when getting used to not having 100% coverage led to not notice some areas which should really be covered.

Reverted it.